### DISCUSSION TITLE
Json to nested enum

### DISCUSSION THREAD

**Discussion by neilkachu** [SwiftGen/SwiftGen#853] [Q&A]
Hi,

Im looking to implement this but have some future use mind too. Currently I'm using basic strings for colours/localisation config but would like to use json to improve readability in use but also future proof this data for any api endpoint.
So..., I have constructed the json like so:

`{
  "core": {
    "primary": "#000000",
    "secondary": "#e8e8e8",
    "tertiary": "#FFFFFF",
    "text": "#000000"
  },
  "button": {
    "label": "#000000",
    "background": "#FFFFFF"
  }}`

Id like this to be retrievable after conversion as a nested enum ie `Colors.core.primary` as is available via the strings templates , is this possible? The existing json templates only produce a single level Dictionary entry.

Cheers, Neil

**Comment by djbe** [comment]
Quite a similar question to https://github.com/SwiftGen/SwiftGen/discussions/850. As I mentioned there:

The sample templates for JSON, YAML and Plists are only just that, example templates. They work for very simple use cases, but will not match what many users need.

They (the built-in templates) definitely cannot generate the code you're proposing, because they have to handle much more general cases. 
- I assume you're starting from the `inline-swift5` template, [docs for it here](https://github.com/SwiftGen/SwiftGen/blob/stable/Documentation/templates/json/inline-swift5.md).
- From that template you can easily create a custom template that matches your needs. You can find an article that [describes this process here](https://github.com/SwiftGen/SwiftGen/blob/stable/Documentation/Articles/Creating-custom-templates.md).
- In your case, you could add some code to parse the hex strings to colors, or invoke a color init you have somewhere else.

**Reply by neilkachu** [reply]
Cheers Dave,
Yes, Ive looked at the json and strings templates and done some changes to ensure key/enum case compatibility already so I'll delve deeper, as long as it is actually doable I'll see what I can come up with by merging these 2 types. 
This would seem to be a common use case as strings files are a pain and not compatible server side for cross platform work.

Thanks for your assistance!
